ci/ci.go

```diff
@@ -11,5 +14,5 @@ type Job struct {
 // Runner contains methods for running user provided code in isolation.
 type Runner interface {
 	// Run should synchronously execute the described job and return the output.
-	Run(context.Context, *Job, string) (string, error)
+	Run(context.Context, *Job, string, time.Duration) (string, error)
```

Comment:
Can't we use the `context.Context` to handle the timeout? 
https://golang.org/pkg/context/#example_WithTimeout

Reply:
Technically, sure. But do you mean passing the context with an already set timeout to the `Run()` method? 
The problem with this is (just tried it) that all the Docker API methods use the same context. Because of that, when the context timeout is reached, container just fails and `Run()` returns with an error about this:
`ERROR   ci/run_tests.go:62      Test execution failed: %!w(context.deadlineExceededError={})`
I.e., there will be a log message to debug, but no new submission with updated build log will be saved in the database, and therefore no information about what happened will be available to teachers or students.

Reply:
Don't have time to investigate this now, but I was thinking that we still control the timeout of the context. I'm not sure all of them need to use the same context, but even so, why would they timeout with a X minute long timeout. (Will try to understand more later)